test/unit/math/prim/fun/uniform_simplex_test.cpp

```diff
@@ -0,0 +1,31 @@
+#include <stan/math/prim.hpp>
+#include <gtest/gtest.h>
+
+TEST(MathFunctions, uniform_simplex) {
+  using Eigen::VectorXd;
+  using stan::math::uniform_simplex;
+
+  VectorXd u0 = uniform_simplex(0);
```

Comment:
This should throw, not return a size zero result, because a size zero vector is not a simplex (entries don't sum to 1).